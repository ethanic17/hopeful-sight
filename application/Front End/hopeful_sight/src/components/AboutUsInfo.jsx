import aboutphoto from "../pages/AboutUsImages/kidwithglasses.jpg";

export function AboutUsInfo() {
  return (
    <div className="bg-white p-4 rounded-lg w-3/5 h-[700px] shadow-md mx-auto">
      <div>
        <img
          src={aboutphoto}
          alt="photo"
          className="object-cover w-full h-full rounded-lg bg-transparent"
        />
      </div>
      <br />
      <h1 className="text-2xl text-center font-semibold">
        Helping the world one pair at a time
      </h1>

      <hr className="border-t-2 my-4 w-1/2 mx-auto border-blue-500" />
      <h2 className="text-xl text-center">
        Hopeful Sight is a non-profit and nonpartisan organization that helps
        the less fortunate acquire life altering vision correcting glasses.
      </h2>
    </div>
  );
}
