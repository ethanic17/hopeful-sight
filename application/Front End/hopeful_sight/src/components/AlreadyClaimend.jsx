import { FaCheckCircle } from "react-icons/fa";

export function AlreadyClaimed() {
  return (
    <div className="bg-gray-50 rounded-lg shadow-lg p-8 max-w-lg mx-auto text-center">
      <div className="mb-6 text-green-500">
        <FaCheckCircle className="w-16 h-16 mx-auto" />
      </div>

      <h2 className="text-3xl font-semibold text-gray-800 mb-4">
        Your Glasses Are on Their Way!
      </h2>

      <p className="text-lg text-gray-700 mb-6">
        You've already claimed your pair of glasses! Weâ€™re thrilled to support
        you on your journey to clearer vision.
      </p>

      <p className="text-md text-gray-600 mb-6">
        Thank you to our generous donors who made this possible. Your
        contribution is making a meaningful impact in someone's life, and we
        couldn't do this without you!
      </p>

      <p className="text-sm text-gray-500 mb-6">
        If you need further assistance or have any questions, don't hesitate to
        reach out to us. We're here to help!
      </p>

      <button className="bg-sky-500 hover:bg-sky-600 text-white font-medium py-2 px-4 rounded-lg transition-all">
        Contact Support
      </button>
    </div>
  );
}
