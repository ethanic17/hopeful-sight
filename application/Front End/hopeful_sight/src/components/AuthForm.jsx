import { useState } from "react";
import { SignInForm } from "./SignInForm";
import { SignUpForm } from "./SignUpForm";

export function AuthForm() {
  const [hasAccount, setHasAccount] = useState(true);

  return (
    <div className="w-full h-full flex justify-center items-center">
      <div className="w-1/2">
        {hasAccount ? (
          <SignInForm setHasAccount={setHasAccount} />
        ) : (
          <SignUpForm setHasAccount={setHasAccount} />
        )}
      </div>
    </div>
  );
}
