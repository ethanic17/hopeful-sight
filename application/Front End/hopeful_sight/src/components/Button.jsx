export function Button({ onClick, children, className = "" }) {
  return (
    <button
      onClick={onClick}
      className={`bg-blue-500 hover:bg-blue-700 text-white font-light py-2 px-4 rounded text-xl  ${className}`}
    >
      {children}
    </button>
  );
}
