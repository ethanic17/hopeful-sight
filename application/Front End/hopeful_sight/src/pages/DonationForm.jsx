/**
 * DonationForm component for handling user donations and sign-ups
 * This form collects donation amount and user details like name, email, phone, and address
 * Once submitted, it navigates to a confirmation page
 */
import React, { useState } from "react";
import { useNavigate } from "react-router-dom";

/**
 * @typedef {Object} FormData
 *
 * The user's name
 * @property {string} name
 *
 * The user's email
 * @property {string} email
 *
 * The user's phone number in the format 100-100-1000
 * @property {string} phone
 *
 * The user's address
 * @property {string} address
 */

/**
 * The JSX form for donation and sign-up
 * @returns {JSX.Element}
 */
function DonationForm() {
  const [amount, setAmount] = useState("");
  const [formData, setFormData] = useState({
    name: "",
    email: "",
    phone: "",
    address: "",
  });
  const navigate = useNavigate();

  /**
   * Updates the form data state when an input field changes
   *
   * The change event
   * @param {React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>} e
   */
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((prevState) => ({
      ...prevState,
      [name]: value,
    }));
  };

  /**
   * Handles form submission, prevents default action and navigates to confirmation
   *
   * The form submission event
   * @param {React.FormEvent} e
   */
  const handleSubmit = (e) => {
    e.preventDefault();
    // We have the option to redirect this submission to the api
    navigate("/confirmation");
  };

  return (
    <div className="max-w-md mx-auto mt-10">
      <form
        onSubmit={handleSubmit}
        className="bg-white shadow-lg rounded-lg p-6"
      >
        <h2 className="text-xl font-bold text-gray-800 mb-6 text-center">
          Donation and Sign Up Form
        </h2>

        {/* Donation amount */}
        <div className="mb-4">
          <label
            htmlFor="amount"
            className="block text-gray-800 text-sm font-bold mb-3"
          >
            Donation Amount:
          </label>
          <input
            type="number"
            value={amount}
            onChange={(e) => setAmount(e.target.value)}
            placeholder="Enter Donation Amount"
            id="amount"
            name="amount"
            className="shadow appearance-none border rounded w-full py-3 px-3 text-gray-900 leading-tight focus:outline-none focus:shadow-outline"
          />
        </div>

        {/* Name */}
        <div className="mb-4">
          <label
            htmlFor="name"
            className="block text-gray-800 text-sm font-bold mb-3"
          >
            Name:
          </label>
          <input
            type="text"
            id="name"
            name="name"
            value={formData.name}
            onChange={handleChange}
            required
            className="shadow appearance-none border rounded w-full py-3 px-3 text-gray-900 leading-tight focus:outline-none focus:shadow-outline"
          />
        </div>

        {/* Email */}
        <div className="mb-4">
          <label
            htmlFor="email"
            className="block text-gray-800 text-sm font-bold mb-3"
          >
            Email:
          </label>
          <input
            type="email"
            id="email"
            name="email"
            value={formData.email}
            onChange={handleChange}
            required
            className="shadow appearance-none border rounded w-full py-3 px-3 text-gray-800 leading-tight focus:outline-none focus:shadow-outline"
          />
        </div>

        {/* Phone */}
        <div className="mb-4">
          <label
            htmlFor="phone"
            className="block text-gray-800 text-sm font-bold mb-3"
          >
            Phone Number (100-100-1000):
          </label>
          <input
            type="tel"
            id="phone"
            name="phone"
            pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
            value={formData.phone}
            onChange={handleChange}
            required
            className="shadow appearance-none border rounded w-full py-3 px-3 text-gray-800 leading-tight focus:outline-none focus:shadow-outline"
          />
        </div>

        {/* Address */}
        <div className="mb-6">
          <label
            htmlFor="address"
            className="block text-gray-800 text-sm font-bold mb-3"
          >
            Address:
          </label>
          <textarea
            id="address"
            name="address"
            value={formData.address}
            onChange={handleChange}
            required
            className="shadow appearance-none border rounded w-full py-3 px-3 text-gray-800 leading-tight focus:outline-none focus:shadow-outline h-10 resize-none"
          ></textarea>
        </div>

        {/* Submit button */}
        <button
          type="submit"
          className="w-full bg-blue-500 hover:bg-blue-800 text-white font-bold py-3 px-4 rounded focus:outline-none focus:shadow-outline"
        >
          Submit Donation
        </button>
      </form>
    </div>
  );
}

export default DonationForm;
