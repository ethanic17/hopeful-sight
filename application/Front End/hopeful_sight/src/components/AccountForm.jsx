import { FormBody } from "./FormBody";
import { Button } from "./Button";
import { Input } from "./Input";
import { useState } from "react";
import useAxiosWithToken from "../hooks/axios";
import { useNavigate } from "react-router-dom";
import { useDispatch } from "react-redux";
import {
  addAccountID,
  addAccountType,
  addAccountTypeID,
} from "../app/features/userSlice";

export function AccountForm({ accountType, setStep, userID }) {
  const [phoneNumber, setPhoneNumber] = useState("");
  const [address, setAddress] = useState("");
  const [city, setCity] = useState("");
  const [state, setState] = useState("");
  const [zipCode, setZipCode] = useState("");
  const [loading, setLoading] = useState(false);
  const axiosInter = useAxiosWithToken();
  const dispatch = useDispatch();
  const navigate = useNavigate();
  console.l;
  async function submitForm(e) {
    e.preventDefault();
    setLoading(true);
    let resp = await axiosInter.post("/api/accounts/", {
      phone_number: phoneNumber,
      address: address,
      city: city,
      state: state,
      zip_code: zipCode,
      user: userID,
    });
    if (resp.status === 201) {
      console.log(resp.data);
      dispatch(addAccountID(resp.data.account_id));
      if (accountType === "donatee") {
        setStep((state) => state + 1);
      } else {
        dispatch(addAccountType(accountType));
        await createDonator(resp.data.account_id);
        navigate("/account");
      }
    }

    setLoading(false);
  }

  async function createDonator(id) {
    let resp = await axiosInter.post("/api/donators/", {
      has_donated: false,
      total_amount_donated: 0,
      account: id,
    });
    console.log(resp);
    dispatch(addAccountTypeID(resp.donator_id));
  }

  return (
    <FormBody>
      <div className="flex space-x-2">
        <div className="h-4 w-4 rounded-full bg-blue-800"></div>
        <div className="h-4 w-4 rounded-full bg-blue-800"></div>
        {accountType == "donatee" && (
          <div className="h-4 w-4 rounded-full bg-gray-800"></div>
        )}
      </div>
      <div className="flex flex-col space-y-4 w-full">
        <Input
          state={phoneNumber}
          setState={setPhoneNumber}
          placeholder="Phone Number"
          type="text"
        />

        <Input
          state={address}
          setState={setAddress}
          placeholder="Address"
          type="text"
        />

        <Input state={city} setState={setCity} placeholder="City" type="text" />

        <Input
          state={state}
          setState={setState}
          placeholder="State"
          type="text"
        />

        <Input
          state={zipCode}
          setState={setZipCode}
          placeholder="Zip Code"
          type="text"
        />
      </div>

      <Button onClick={submitForm}>{loading ? "Loading..." : "Submit"}</Button>
      <p
        className="w-full"
        onClick={(e) => {
          e.preventDefault();
          setStep((state) => state - 1);
        }}
      >
        Back
      </p>
    </FormBody>
  );
}
