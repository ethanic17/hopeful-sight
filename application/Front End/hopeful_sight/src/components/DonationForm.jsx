import { useState } from "react";
import { useDispatch } from "react-redux";
import { addDonationAmount } from "../app/features/userSlice";

import { Button } from "./Button";
import { FormBody } from "./FormBody";
import { Input } from "./Input";
import { useNavigate } from "react-router-dom";

export function DonationForm() {
  const [donationAmount, setDonationAmount] = useState("");
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleDonation = (e) => {
    e.preventDefault();
    dispatch(addDonationAmount(donationAmount));
    navigate("/account");
  };

  return (
    <FormBody>
      <h3 className="text-center mb-4 text-lg font-semibold">
        Enter an amount to donate
      </h3>
      <Input
        type="number"
        placeholder="Enter donation amount"
        state={donationAmount}
        setState={setDonationAmount}
      />
      <Button onClick={handleDonation}>Donate</Button>
    </FormBody>
  );
}
