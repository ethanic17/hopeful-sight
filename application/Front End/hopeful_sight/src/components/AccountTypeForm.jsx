import { AccountForm } from "./AccountForm";
import { Button } from "./Button";
import { FormBody } from "./FormBody";
import { DonateeForm } from "./DonateeForm";
import { useState } from "react";

export function AccountTypeForm() {
  const [accountType, setAccountType] = useState("none");
  const [step, setStep] = useState(0);
  switch (step) {
    case 0:
      return (
        <FormBody>
          <h3 className="text-2xl font-bold mb-4">Select Account Type:</h3>
          <div className="flex justify-around flex-row gap-x-5">
            <Button
              onClick={(e) => {
                setAccountType("donator");
                e.preventDefault();
                setStep(step + 1);
              }}
            >
              As Donator
            </Button>
            <Button
              onClick={(e) => {
                e.preventDefault();
                setStep(step + 1);
                setAccountType("donatee");
              }}
            >
              As Donatee
            </Button>
          </div>
        </FormBody>
      );
    case 1:
      return (
        <AccountForm
          back={setStep}
          setStep={setStep}
          accountType={accountType}
        />
      );
    case 2:
      return <DonateeForm />;
  }
}
