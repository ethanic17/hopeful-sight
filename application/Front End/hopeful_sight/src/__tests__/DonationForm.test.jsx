import { vi, describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { DonationForm } from "../components/DonationForm";
import { useState } from "react";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import mockStore from "./mockStore";

vi.mock("react", async () => {
  const actual = await vi.importActual("react");
  return {
    ...actual,
    useState: vi.fn(),
  };
});

describe("DonationForm", () => {
  const mockUseState = vi.mocked(useState);

  it("renders step 0 (PaymentForm)", () => {
    mockUseState.mockReturnValue([0, vi.fn()]);
    render(
      <BrowserRouter>
        <DonationForm currentGasses={[]} />
      </BrowserRouter>,
    );
    expect(
      screen.getByText("Transform Lives with the Gift of Sight"),
    ).toBeInTheDocument();
  });

  it("renders step 1 (DonationAmount)", () => {
    mockUseState.mockReturnValue([1, vi.fn()]);
    render(
      <Provider store={mockStore(true, false)}>
        <BrowserRouter>
          <DonationForm currentGasses={[]} />
        </BrowserRouter>
      </Provider>,
    );
    expect(screen.getByText("Quick Donation Amounts")).toBeInTheDocument();
  });

  it("renders step 2 (DonationConfirm)", () => {
    mockUseState.mockReturnValue([2, vi.fn()]);
    render(
      <BrowserRouter>
        <DonationForm currentGasses={[]} />
      </BrowserRouter>,
    );
    expect(
      screen.getByText("Your donation has been received successfully."),
    ).toBeInTheDocument();
  });
});
