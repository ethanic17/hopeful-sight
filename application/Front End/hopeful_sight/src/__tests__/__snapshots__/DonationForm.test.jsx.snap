// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DonationForm > renders correctly 1`] = `
<div>
  <div
    class="flex h-full items-center justify-center bg-gray-100 p-10 space-x-10"
  >
    <div
      class="w-1/2 bg-gradient-to-br from-sky-600 to-blue-900 text-white p-10 rounded-xl shadow-lg hover:shadow-2xl"
    >
      <h2
        class="text-2xl font-bold mb-4 flex items-center gap-2"
      >
        Transform Lives with the Gift of Sight
      </h2>
      <p
        class="text-lg leading-relaxed mb-6"
      >
        Over 
        <strong>
          800 million people
        </strong>
         lack access to glasses, limiting their ability to learn, work, and thrive. With
         
        <strong>
          any amount you’re able to give
        </strong>
        , you can provide someone with the gift of clear sight, unlocking a brighter future for themselves and their community.
      </p>
      <em
        class="block text-base font-medium"
      >
        Your support changes lives—one lens at a time. ❤️
      </em>
    </div>
    <div
      class="w-1/2"
    >
      <form
        class="bg-gradient-to-r from-blue-100 to-gray-200 p-7 rounded-xl w-full flex justify-center items-center flex-col space-y-10 shadow-lg"
      >
        <div
          class="w-2/3"
        >
          <h1
            class="text-center mb-6 text-4xl font-extrabold bg-gradient-to-r from-sky-500 to-sky-600 bg-clip-text text-transparent"
          >
            Support Hopeful Sight by Donating
          </h1>
          <div
            class="my-5 w-full"
          >
            <div
              class="relative"
            >
              <label
                class="absolute left-3 z-10 transition-all duration-200 top-3 text-gray-500  pl-10"
                for="name on card"
              >
                Name on card
                <span
                  class="text-red-600 ml-1"
                >
                  *
                </span>
              </label>
              <div
                class="relative"
              >
                <svg
                  class="absolute left-3 top-1/2 -translate-y-1/2 text-gray-400"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M4 22C4 17.5817 7.58172 14 12 14C16.4183 14 20 17.5817 20 22H18C18 18.6863 15.3137 16 12 16C8.68629 16 6 18.6863 6 22H4ZM12 13C8.685 13 6 10.315 6 7C6 3.685 8.685 1 12 1C15.315 1 18 3.685 18 7C18 10.315 15.315 13 12 13ZM12 11C14.21 11 16 9.21 16 7C16 4.79 14.21 3 12 3C9.79 3 8 4.79 8 7C8 9.21 9.79 11 12 11Z"
                  />
                </svg>
                <input
                  class="name on card w-full align-center rounded-md border border-gray-300 py-2 px-3 pt-4 shadow-sm transition-all duration-200 focus:border-blue-500 focus:outline-none focus:ring-1 focus:ring-blue-500 pl-10 undefined"
                  data-testid="name on card"
                  placeholder=""
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="my-5 w-full"
          >
            <div
              class="relative"
            >
              <label
                class="absolute left-3 z-10 transition-all duration-200 top-3 text-gray-500  pl-10"
                for="email"
              >
                Email
                <span
                  class="text-red-600 ml-1"
                >
                  *
                </span>
              </label>
              <div
                class="relative"
              >
                <svg
                  class="absolute left-3 top-1/2 -translate-y-1/2 text-gray-400"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 3H21C21.5523 3 22 3.44772 22 4V20C22 20.5523 21.5523 21 21 21H3C2.44772 21 2 20.5523 2 20V4C2 3.44772 2.44772 3 3 3ZM20 7.23792L12.0718 14.338L4 7.21594V19H20V7.23792ZM4.51146 5L12.0619 11.662L19.501 5H4.51146Z"
                  />
                </svg>
                <input
                  class="email w-full align-center rounded-md border border-gray-300 py-2 px-3 pt-4 shadow-sm transition-all duration-200 focus:border-blue-500 focus:outline-none focus:ring-1 focus:ring-blue-500 pl-10 undefined"
                  data-testid="email"
                  placeholder=""
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="my-5 w-full"
          >
            <div
              class="relative"
            >
              <label
                class="absolute left-3 z-10 transition-all duration-200 top-3 text-gray-500  pl-10"
                for="card number"
              >
                Card Number
                <span
                  class="text-red-600 ml-1"
                >
                  *
                </span>
              </label>
              <div
                class="relative"
              >
                <svg
                  class="absolute left-3 top-1/2 -translate-y-1/2 text-gray-400"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3.00488 2.99979H21.0049C21.5572 2.99979 22.0049 3.4475 22.0049 3.99979V19.9998C22.0049 20.5521 21.5572 20.9998 21.0049 20.9998H3.00488C2.4526 20.9998 2.00488 20.5521 2.00488 19.9998V3.99979C2.00488 3.4475 2.4526 2.99979 3.00488 2.99979ZM20.0049 10.9998H4.00488V18.9998H20.0049V10.9998ZM20.0049 8.99979V4.99979H4.00488V8.99979H20.0049ZM14.0049 14.9998H18.0049V16.9998H14.0049V14.9998Z"
                  />
                </svg>
                <input
                  class="card number w-full align-center rounded-md border border-gray-300 py-2 px-3 pt-4 shadow-sm transition-all duration-200 focus:border-blue-500 focus:outline-none focus:ring-1 focus:ring-blue-500 pl-10 undefined"
                  data-testid="card number"
                  placeholder=""
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div
            class="flex justify-end w-full"
          >
            <button
              class="bg-blue-500 hover:bg-blue-700 text-white font-light py-2 px-4 rounded text-xl"
            >
              Next
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
